File: graviton - game/src/main/resources/scripts/commands/fight.js
importClass(org.graviton.database.repository.CommandRepository);
importClass(org.graviton.game.command.api.AbstractCommand);
importClass(java.util.ArrayList);
importClass(org.graviton.game.client.player.Player);
importClass(org.graviton.game.creature.monster.Monster);
importClass(org.graviton.game.creature.monster.MonsterGroup);

commandRepository.register(JavaAdapter(AbstractCommand, {
    name: function () {
        return "fight";
    },

    description: function () {
        return "";
    },

    apply: function (player, data) {
        var monsters = new ArrayList();

        data[1].split(',').forEach(function(entry) {
            var monster = player.getEntityFactory().getMonsterTemplate(parseInt(entry));

            if(monster != null)
                monsters.add(monster.getRandom().copy());

        });

        var monsterGroup = new MonsterGroup(0, player.getGameMap(), 0, monsters);

        player.getGameMap().getFightFactory().newMonsterFight(player, monsterGroup);
    }

}));

File: graviton - game/src/main/java/org/graviton/game/action/npc/Fight.java
package org.graviton.game.action.npc;

import org.graviton.game.action.Action;
import org.graviton.game.action.common.GameAction;
import org.graviton.game.creature.monster.Monster;
import org.graviton.game.creature.monster.MonsterGroup;
import org.graviton.game.creature.monster.MonsterTemplate;
import org.graviton.network.game.GameClient;

import java.util.Collections;

/**
 * Created by Botan on 14/07/17. 23:54
 */

@GameAction(id = 544)
public class Fight implements Action{

    @Override
    public void apply(GameClient client, Object data) {
        MonsterTemplate monsterTemplate = client.getEntityFactory().getMonsterTemplate(Short.parseShort(((String) data).split(",")[0]));
        Monster monster = monsterTemplate.getByLevel(Short.parseShort(((String) data).split(",")[1])).copy();

        MonsterGroup monsterGroup = new MonsterGroup(0, client.getPlayer().getGameMap(), (short) 0, Collections.singleton(monster));

        client.getPlayer().getGameMap().getFightFactory().newMonsterFight(client.getPlayer(), monsterGroup, false);
    }

    @Override
    public void finish() {

    }
}
